querymiddleware: close MQE queries on every error path (#15251)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] MQE: Add support for common subexpression elimination and subset selector elimination of range vector selectors in range queries. Enable with `-querier.mimir-query-engine.enable-range-query-range-vector-common-subexpression-elimination=true`. #15127
* [ENHANCEMENT] MQE: Use series selected for one side to reduce data selected on the other side in one-to-many and many-to-one binary operations (eg. `group_left` and `group_right`). #15137
* [ENHANCEMENT] MQE: Simplify `unless` and `or` operations where one side can be proven to be empty by inspecting the expression. #15198
* [ENHANCEMENT] MQE: `MemoryConsumptionTracker` no longer retains the request `context.Context`. #15251

> * [BUGFIX] Query-frontend: Close MQE queries on all error paths to prevent leaking memory tracker registrations and pooled buffers. #15251

Reviewer: Can you please phrase these in terms a user will understand?

Author: https://github.com/grafana/mimir/pull/15251/changes/69a3bc404126c8bd23e933a8404ac606a5133644

> * [BUGFIX] Query-frontend: Fixed a MemoryConsumptionTracker memory leak caused by MQE queries not having `Close()` called in some error paths. #15251

Reviewer: `MemoryConsumptionTracker` and `Close` mean nothing to a user - how about this?

```suggestion
* [BUGFIX] Query-frontend: Fixed a memory leak caused that could occur on some error paths if MQE was enabled. #15251
```
